sentry_sdk/scrubber.py

```diff
@@ -66,7 +66,7 @@ def __init__(self, denylist=None, recursive=False):
         self.recursive = recursive
 
     def scrub_list(self, lst):
-        # type: (List[Any]) -> None
+        # type: (object) -> None
```

Comment:
I get where you're coming from (the method theoretically does support any type), but as a user I'd find it confusing that a method called `scrub_list` accepts an argument `lst` of any type.

Since types are not enforced at runtime, users are free to ignore the type hint that says a `List[Any]` should be given to this method, which is why the `isinstance(lst, list)` check is there now -- just to make sure we don't explode if someone misuses the function. I feel like by changing the type hint to say this function supports any `object`, we're making it harder for users to understand how this method is meant to be used.

Reply:
I need this change for the Event TypedDict changes (#2753). Originally, I was going to include these changes there, but I split these into a separate PR, since they are not strictly related.

We should use the `object` type here because without this change, the `sentry_sdk.scrubber.EventScrubber.scrub_request` method raises type errors on the #2753 code. The reason is that `event["request"]` has a type of `dict[str, object]`, since the request object (at least from my understanding) can contain an arbitrary mapping from strings to any object. We are then calling `scrub_dict` on several objects within `event["request"]` (such as `event["request"]["headers"]`), which according to the type checker have type `object`. However without this PR, the `scrub_dict` method is declared as only accepting `dict[str, Any]`, so the type checker fails, since based on the type hint it appears that the code is not type safe.

However, because of the fact that we are doing the `isinstance` check within the `scrub_dict` method, the code is in fact type safe. We can pass any `object` to `scrub_dict` and the code will behave correctly.

> I feel like by changing the type hint to say this function supports any object, we're making it harder for users to understand how this method is meant to be used.

I understand your point, but there is already a doc comment here to explain that the method only does anything if the argument passed is a list, and I will also add a similar comment to the `scrub_dict` method to clarify.

I prefer `object` here, since in my opinion, the purpose of type hints is to communicate to users what type they need to pass in order for type safety to be ensured. For these methods, users can pass objects even with an unknown type without violating the method contract, and type safety is always guaranteed. Going back to the `scrub_request` example that prompted me to make this change, having type `object` makes it clear that I can safely pass `event["request"]["headers"]` without first adding an `isinstance` check to make sure that `event["request"]["headers"]` is a dictionary.

Reply:
Gotcha -- then let's go with your change.